src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -2352,6 +2424,21 @@ void Llvm::buildCkFinite(GenTreeUnOp* ckNode)
     mapGenTreeToValue(ckNode, opValue);
 }
 
+void Llvm::buildKeepAlive(GenTreeUnOp* keepAliveNode)
+{
+    // KEEPALIVE is used to represent implicit uses of GC-visible values, e. g.:
+    //
+    //  ObjWithFinalizer obj = new ObjWithFinalizer();
+    //  NativeResource handle = obj.NativeResource;
+    //  <-- Here the compiler could think liveness of "obj" ends and permit its finalization. -->
+    //  NativeCall(handle);
+    //  <-- We insert KeepAlive s.t. we don't finalize away "handle" while it is still in use by the native call. -->
+    //  GC.KeepAlive(obj)
+    //
+    // In the shadow stack model, this would have to be done before codegen, by spilling "obj", so we don't need to do
```

Comment:
Is it worth putting a TODO in llvmcodegen somewhere about this?

Reply:
I suppose it is already handled in lowering save for the SDSU case we discussed some time back. I will reword this a little.